src/qibolab/instruments/qblox/controller.py

```diff
@@ -40,46 +41,50 @@ def connect(self):
         if self.is_connected:
             return
         try:
-            self.cluster.connect()
-            for name in self.modules:
-                self.modules[name].connect()
+            # Connect cluster
+            QbloxCluster.close_all()
+            self.cluster = QbloxCluster(self.name, self.address)
+            self.cluster.reset()
+            self.cluster.set("reference_source", self._reference_clock)
+            # Connect modules
+            for module in self.modules.values():
+                module.connect(self.cluster)
             self.is_connected = True
+            log.info("QbloxController: all modules connected.")
+            return
+
         except Exception as exception:
-            raise_error(
-                RuntimeError,
-                "Cannot establish connection to " f"{self.modules[name]} module. " f"Error captured: '{exception}'",
-            )
+            raise ConnectionError(f"Unable to connect:\n{str(exception)}\n")
             # TODO: check for exception 'The module qrm_rf0 does not have parameters in0_att' and reboot the cluster
 
-        else:
-            log.info("QbloxController: all modules connected.")
-
     def setup(self):
-        """Sets all modules up."""
+        """Empty method to comply with Instrument interface.
+        Setup of the modules happens in the create method:
 
-        if not self.is_connected:
-            raise_error(
-                RuntimeError,
-                "There is no connection to the modules, the setup cannot be completed",
-            )
-        self.cluster.setup()
-        for name in self.modules:
-            self.modules[name].setup()
+        >>> instruments = load_instrument_settings(runcard, instruments)
+        """
+        pass
+
+        # if not self.is_connected:
+        #     raise_error(
+        #         RuntimeError,
+        #         "There is no connection to the modules, the setup cannot be completed",
+        #     )
+        # for name in self.modules:
+        #     #THIS is
+        #     self.modules[name].setup()
```

Comment:
Maybe we can remove this
```suggestion
```
but I will probably drop these methods for all instruments in another PR (when fixing #673).